src/android/PurchasesPlugin.java

```diff
@@ -179,6 +181,44 @@ private void isAnonymous(JSONArray productIDs, CallbackContext callbackContext)
         Map<String, Map<String, Object>> map = CommonKt.checkTrialOrIntroductoryPriceEligibility(productIDList);
         callbackContext.success(convertMapToJson(map));
     }
+    
+    @PluginAction(thread = ExecutionThread.WORKER, actionName = "invalidatePurchaserInfoCache")
+    private void invalidatePurchaserInfoCache(CallbackContext callbackContext) {
+        CommonKt.invalidatePurchaserInfoCache();
+    }
+
+    //================================================================================
+    // Subscriber Attributes
+    //================================================================================
+
+    @PluginAction(thread = ExecutionThread.WORKER, actionName = "setAttributes")
+    private void setAttributes(JSONObject attributes, CallbackContext callbackContext) throws JSONException {
+        CommonKt.setAttributes(convertJsonToMap(attributes));
+    }
+
+    @PluginAction(thread = ExecutionThread.WORKER, actionName = "setEmail")
+    private void setEmail(String email, CallbackContext callbackContext) {
+        CommonKt.setEmail(email);
+    }
+
+    @PluginAction(thread = ExecutionThread.WORKER, actionName = "setPhoneNumber")
+    private void setPhoneNumber(String phoneNumber, CallbackContext callbackContext) {
+        CommonKt.setPhoneNumber(phoneNumber);
+    }
+
+    @PluginAction(thread = ExecutionThread.WORKER, actionName = "setDisplayName")
+    private void setDisplayName(String displayName, CallbackContext callbackContext) {
+        CommonKt.setDisplayName(displayName);
+    }
+
+    @PluginAction(thread = ExecutionThread.WORKER, actionName = "setPushToken")
+    private void setPushToken(String pushToken, CallbackContext callbackContext) {
+        CommonKt.setPushToken(pushToken);
```

Comment:
This is interesting. The `callbackContext` is not notified by doing `callbackContext.success()`. But if you check the `setDebugLogsEnabled` we are not doing it either. If you check the iOS code, we do notify tho....We should check what's the right thing to do

Reply:
I can't find good docs on it. It seems cleaner to call success, though. I'll add it